src/Common/Core/Impl/Logging/Implementation/ServiceLogger.cs

```diff
@@ -7,12 +7,16 @@
 using Microsoft.Common.Core.Disposables;
 using Microsoft.Extensions.Logging;
 
-namespace Microsoft.R.Host.UserProfile {
-    internal sealed class ServiceLogger : ILogger, IDisposable {
+namespace Microsoft.Common.Core.Logging {
+    public sealed class ServiceLogger : ILogger, IDisposable {
         private readonly string _category;
+        private readonly LogLevel _logLevel;
+        private readonly string _eventLogSource;
 
-        public ServiceLogger(string category) {
+        public ServiceLogger(string category, LogLevel minLogLevel, string eventLogSource) {
```

Comment:
I added this setting so that RUserProfileService can log at `Trace` level and Broker can log at `Warning` level. Since RUserProfileService only logs via the event logs, it should log everything via the event logs. For broker, all the log messages go into the log file, and anything at warning or greater level goes into event logs.